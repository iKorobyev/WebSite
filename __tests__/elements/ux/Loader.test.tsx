import React from 'react'
import { cleanup, render, waitFor, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Loader } from '~ux'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Listbox', () => {
  it('renders correctly', async () => {
    const props = {}

    const { container } = render(<Loader {...props}>Test</Loader>)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })
})
