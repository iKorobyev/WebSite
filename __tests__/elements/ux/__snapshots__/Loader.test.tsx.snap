// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listbox renders correctly 1`] = `
<div>
  <div
    class=""
    data-testid="loader"
  >
    <div
      aria-busy="true"
      class=""
    >
      <svg
        aria-label="audio-loading"
        height="60"
        stroke="green"
        viewBox="0 0 38 38"
        width="60"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            stroke-width="2"
            transform="translate(1 1)"
          >
            <circle
              cx="18"
              cy="18"
              r="18"
              stroke-opacity=".5"
            />
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
